api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -90,12 +90,15 @@ public AllergyIntolerance toFhirResource(Allergy omrsAllergy) {
 	
 	@Override
 	public Allergy toOpenmrsType(AllergyIntolerance fhirAllergy) {
+		if (fhirAllergy == null) {
+			return null;
+		}
```

Comment:
Is this, strictly speaking, necessary?

Reply:
Yes, because if fhirAllergy is null and we are not returning null here or throwing an exception, it would lead to a new instance of Allergy being returned.

Reply:
Oh right! I missed that! Cheerfully withdrawn!